Document that -safe-string is the default, -unsafe-string is not.

pmetzger opened the pull request:
Note that -safe-string is the default, as it has been since 4.06.
Similarly, remove the annotation that -unsafe-string is the default.

gasche commented:
Could you also do the same change in the manual, that is `manual/manual/cmds/unified-options.etex`?

(Is it intentional to not write a Changes entry?)

pmetzger commented:
I didn't write a Changes entry because it doesn't seem like it's a real change. I'm simply documenting what already is true. If you feel I need to, though, I will.

I'll make the same change in the manual, I hadn't thought about doing that one.

pmetzger commented:
By the way, I think this should be cherry-picked for 4.07.

damiendoligez commented on man/ocaml.m:
```diff
@@ -207,9 +206,7 @@ Turn bound checking off on array and string accesses (the
-thereby making strings writable. For reasons of backward compatibility,
-this is the default setting for the moment, but this will change in a future
-version of OCaml.
+thereby making strings writable.
```
Maybe add a note that this is for compatibility with old source code and shouldn't be used with new code?

pmetzger replied:
I can easily do that if that's the consensus.

gasche replied:
I also think the suggestion is a good idea, so that would be a consensus among the three of us :-)

pmetzger replied:
Will do it.

pmetzger replied:
Done!

damiendoligez reviewed (approved):
Could be merged as-is, or take my remark into account.

At any rate, this should be cherry-picked to 4.07.

pmetzger committed:
Document that -safe-string is the default, -unsafe-string is not

Nearly identical changes to:
   ocaml.m, ocamlc.m, ocamlopt.m, unified-options.etex

"-safe-string" has been the default since 4.06, so the assertion that
it would be the default someday, and that "-unsafe-string" is still
the default, was incorrect.

damiendoligez commented:
Merged and cherry-picked to 4.07 (commit 5f5b5a97eec24c353c70c7e3ddb7ae9c1febb3c6).
